tools/autograd/derivatives.yaml

```diff
@@ -1411,6 +1411,11 @@
   src: grad.diagonal(offset, dim1, dim2)
   result: auto_linear
 
+- name: as_strided_scatter(Tensor self, Tensor src, int[] size, int[] stride, int? storage_offset=None) -> Tensor
+  self: as_strided_scatter(grad, zeros_like(src), size, stride, storage_offset)
+  src: grad.as_strided(size, stride, storage_offset)
+  result: auto_linear
```

Comment:
The problem is that you assume that doing "as_strided" on the grad with the parameters from the forward will do the right thing (it won't if any of these have different strides/storage_offset to begin with).
Hence the madness of `as_strided_backward()` :D

I think the simplest way to do this is to create a new function in functions manual that call the backward of each elementary function in order (copy_backward, as_strided_backward) as if you were doing AD on the CompositeExplicit implementation that you have. (nit: it's ok to replicate the copy_backward impl from `tensor.cpp` instead of making it a helper function I think).